src/modules/Auth/pages/RecoverPassword/RecoverPassword.tsx

```diff
@@ -0,0 +1,59 @@
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import { Trans } from 'react-i18next';
+import { useEffect } from 'react';
+import { Box } from '@mui/material';
+
+import { RecoverForm } from 'modules/Auth/features/RecoverPassword';
+import { useAsync } from 'shared/hooks';
+import { recoveryLinkHealthCheckApi } from 'api';
+import { StyledBodyLarge, StyledLinkBtn } from 'shared/styles';
+import { Spinner } from 'shared/components';
+import { page } from 'resources';
+
+export const RecoverPassword = () => {
+  const [searchParams] = useSearchParams();
+
+  const navigate = useNavigate();
+
+  const key = searchParams.get('key');
+  const email = searchParams.get('email');
+
+  const {
+    execute: validateKey,
+    isLoading,
+    error,
+  } = useAsync(
+    async (data: { email: string | null; key: string | null }) =>
+      await recoveryLinkHealthCheckApi(data),
+  );
```

Comment:
Granted it doesn't affect functionality, if you navigate to a used link this results in a couple of uncaught exceptions

Reply:
I ended up just calling the function inside a useEffect, and I handled the errors there